analysis/import/SonarImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/sonar/SonarComponentProjectBuilder.kt

```diff
@@ -36,13 +36,15 @@ import de.maibornwolff.codecharta.importer.sonar.model.Qualifier
 import de.maibornwolff.codecharta.model.*
 import de.maibornwolff.codecharta.translator.MetricNameTranslator
 
-class SonarComponentProjectBuilder constructor(
+class SonarComponentProjectBuilder(
         name: String,
         private val sonarCodeURLLinker: SonarCodeURLLinker = SonarCodeURLLinker.NULL,
         private val translator: MetricNameTranslator = MetricNameTranslator.TRIVIAL,
         private val usePath: Boolean = false
 ) {
 
+    private var totalComponents = 0
+    private var processedComponents = -1
```

Comment:
Why not 0? You increment the processed components and then log.
If you processed the first component, you will have 0 / total components processed.

Reply:
Ok the intention here is, that we can also show something along the lines "progess 0/20". In order to do this I set the initial counter to -1, this way we can use the same logic as for all the other increments. Why would we even want to show 0/20? well from my experience it can take quite some time until the first page is obtained (depending on the network), in this case it is useful to see that the process at least started.

Reply:
oh this is the project builder... so forget what i said about the network, it does not apply here... however the counting logic is still correct since we increment the first time when setting the total components. It is however not that important that we do it this way.